woocommerce-gutenberg-products-block.php

```diff
@@ -261,3 +261,54 @@ function woocommerce_blocks_get_php_translation_from_core( $translation, $text,
 }
 
 add_filter( 'gettext', 'woocommerce_blocks_get_php_translation_from_core', 10, 3 );
+
+/**
+ * Check the current WC Blocks path. If the WC Blocks plugin is active but the
+ * current path is from the WC Core, we can consider the plugin is outdated
+ * because JetPack autoloader always loads the newer package.
+ *
+ * @see https://github.com/woocommerce/woocommerce-blocks/issues/5587
+ *
+ * @return bool
+ */
+function woocommerce_blocks_plugin_is_outdated() {
+	$is_active      = false;
+	$active_plugins = get_option( 'active_plugins', array() );
+
+	// We check for the main plugin file only instead of the full path, because
+	// the plugin folder may not be 'woocommerce-blocks'.
+	foreach ( $active_plugins as $plugin ) {
+		if ( strpos( $plugin, 'woocommerce-gutenberg-products-block.php' ) ) {
+			$is_active = true;
+			break;
+		}
+	}
+
+	if ( ! $is_active ) {
+		return false;
+	}
+
+	$woocommerce_blocks_path = \Automattic\WooCommerce\Blocks\Package::get_path();
+
+	if ( strpos( $woocommerce_blocks_path, 'packages/woocommerce-blocks' ) ) {
+		return true;
+	}
+
+	return false;
+}
+
+if ( woocommerce_blocks_plugin_is_outdated() ) {
+	/**
+	 * Add notice to the admin dashboard if the plugin is outdated.
+	 */
+	function woocommerce_blocks_plugin_outdated_notice() {
+		if ( should_display_compatibility_notices() ) {
+			?>
+			<div class="notice notice-warning">
+				<p><?php esc_html_e( 'You have WooCommerce Blocks installed, but the WocCommerce bundled version is running because it is more up-to-date. This may cause unexpected compatibility issues. Please update the WooCommerce Blocks plugin.', 'woo-gutenberg-products-block' ); ?></p>
```

Comment:
Typo:
```suggestion
				<p><?php esc_html_e( 'You have WooCommerce Blocks installed, but the WooCommerce bundled version is running because it is more up-to-date. This may cause unexpected compatibility issues. Please update the WooCommerce Blocks plugin.', 'woo-gutenberg-products-block' ); ?></p>
```